### PR TITLE
Adding Redux

### PR DESCRIPTION
**Pull request by SnakyMz**
Things I have done:

- [ ] Install Redux
- [ ] Configure Redux Store
- [ ] Add bookSlice
- [ ] Add categoriesSlice

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState: [],
+  reducers: {
+    checkStatus: () => 'Under Construction!',
+  },
+});
```
Comment (AbelG101) [inline src/redux/categories/categoriesSlice.js]: - [ ] Ouch! this is a close one but as part of the project requirements you should check the status from the action payload before returning the message in the reducer, this means the reducer should accept `state` and `action` as parameters as show below 👇.
```jsx
reducers: {
    checkStatus: (state, action) => {
      state =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```

**Comment by AbelG101** [comment]
Hello @SnakyMz please restrain from making changes after you request a review as it will alter the outcome of the review and cause confusions

**Comment by SnakyMz** [comment]
> Hello @SnakyMz please restrain from making changes after you request a review as it will alter the outcome of the review and cause confusions

My apologies, I will perform best practices from now.

File: src/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { configureStore } from '@reduxjs/toolkit';
+import booksReducer from './Books/bookSlice';
+
+const store = configureStore({
+  reducer: {
+    books: booksReducer,
+  },
+});
```
Comment (AbelG101) [inline src/redux/store.js]: - [ ] Kindly make sure to add the `categoriesReducer` in your `configureStore` method here just like you did for the `booksReducer`.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @SnakyMz 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Proper use of Redux ✔
- Correct implementation of Reducers and slices ✔
- Correct use of Git Flow ✔
-  Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a good summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (SnakyMz) [reply src/redux/categories/categoriesSlice.js]: Oops! thanks I shall get it now.

File: src/redux/store.js (hunk 2 above)
Comment (SnakyMz) [reply src/redux/store.js]: Thanks will do it now

**Review by Cabraham1 — APPROVED** [review]
Hi @SnakyMz  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**